drivers/npmx/npmx_shell.c

```diff
@@ -1778,6 +1784,164 @@ static int cmd_ldsw_ldo_voltage_get(const struct shell *shell, size_t argc, char
 	return 0;
 }
 
+static int ldsw_soft_start_config_set(const struct shell *shell, size_t argc, char **argv,
+				      ldsw_soft_start_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+	if (argc < 2) {
+		shell_error(shell, "Error: missing LDSW instance index and config value.");
+		return 0;
+	}
+	if (argc < 3) {
+		shell_error(shell, "Error: missing config value.");
+		return 0;
+	}
+
+	int err = 0;
+	uint8_t ldsw_indx = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
+	uint32_t config_val = (uint32_t)shell_strtoul(argv[2], 0, &err);
+
+	if (err != 0) {
```

Comment:
This doesn't check if `ldsw_indx` is integer. `err` is overwritten.
Also, `idx` or `index`.

Reply:
`idx` and `index` will be changed (removed) in other PR.

`shell_strtoul` is not overriding the err value.